include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -249,6 +292,78 @@ class fm_index
         // index.m_sigma = largest_char;
     }
 
+    //!\overload
+    void construct(text_t const & text)
+        //!\cond
+        requires is_collection
+        //!\endcond
+    {
+        // text collection must not be empty
+        if (std::ranges::begin(text) == std::ranges::end(text))
+            throw std::invalid_argument("The text that is indexed cannot be empty.");
+
+        // A text collection that contains the char with rank 255 or 254 cannot be indexed.
+        // 255 will overflow to 0 (reserved by sdsl) and 254 will be incremented to 255 (reserved as delimiter)
+        assert(!std::ranges::any_of(text | ranges::view::join,
+                                    [] (char_type r)
+                                    {
+                                        return to_rank(r) == static_cast<underlying_rank_t<char_type>>(255) ||
+                                               to_rank(r) == static_cast<underlying_rank_t<char_type>>(254);
+                                    }));
+
+        size_t concatenated_size{0};
+        // there must be at least one non-empty text in the collection
+        bool all_empty = true;
+        for (auto && t : text)
+        {
+            if (std::begin(t) != std::end(t))
+            {
+                all_empty = false;
+                concatenated_size += 1 + t.size(); // text size and delimiter
```

Comment:
why do you filter empty sequences? Doesnt this make identifying the right sequence on locate() pretty difficult?